src/frequenz/sdk/timeseries/logical_meter/_formula_steps.py

```diff
@@ -157,6 +160,54 @@ def apply(self, _: List[Optional[float]]) -> None:
         """No-op."""
 
 
+class Averager(FormulaStep):
+    """A formula step for calculating average."""
+
+    def __init__(self, fetchers: List[MetricFetcher]) -> None:
+        """Create an `Averager` instance.
+
+        Args:
+            fetchers: MetricFetchers for the metrics that need to be averaged.
+        """
+        self._fetchers = fetchers
+
+    def __repr__(self) -> str:
+        """Return a string representation of the step.
+
+        Returns:
+            A string representation of the step.
+        """
+        return f"avg({', '.join(repr(f) for f in self._fetchers)})"
+
+    def apply(self, eval_stack: List[Optional[float]]) -> None:
+        """Calculate average of given metrics, push the average to the eval_stack.
+
+        Args:
+            eval_stack: An evaluation stack, to append the calculated average to.
+
+        Raises:
+            FormulaEngineError: when metric fetchers are unable to fetch values.
+        """
+        value_count = 0
+        total = 0.0
+        for fetcher in self._fetchers:
+            next_val = fetcher.value
+            if next_val is None:
+                raise FormulaEngineError(
+                    "Unable to fetch a value from the resampling actor."
+                )
+            if next_val.value is None:
+                continue
+            value_count += 1
+            total += next_val.value
+        if value_count == 0:
+            avg = 0.0
+        else:
+            avg = total / value_count
+
+        eval_stack.append(avg)
```

Comment:
How you know that these values are from the same timestamp?

Reply:
All samples, received in the `MetricFetchers` are resampled with same samplerate and are synced when being fetched the first time [here](https://github.com/frequenz-floss/frequenz-sdk-python/blob/e0bd82f29e2fe55adf0503b8335cd38837371d12/src/frequenz/sdk/timeseries/logical_meter/_formula_engine.py#L123).

Reply:
when using the data pipeline,  people should run a single resampling actor, so that all resampled streams are synchronized.  This should be part of the high level documentation for the data pipeline.